sim/src/mechanics/parking.rs

```diff
@@ -99,6 +99,41 @@ impl ParkingSimState {
         sim
     }
 
+    // Returns any cars that got very abruptly evicted from existence
+    pub fn handle_live_edits(&mut self, map: &Map, timer: &mut Timer) -> Vec<ParkedCar> {
+        let (filled_before, _) = self.get_all_parking_spots();
+        let new = ParkingSimState::new(map, timer);
+        let (_, avail_after) = new.get_all_parking_spots();
+        let avail_after: BTreeSet<ParkingSpot> = avail_after.into_iter().collect();
+
+        // Use the new spots
+        self.onstreet_lanes = new.onstreet_lanes;
+        self.driving_to_parking_lanes = new.driving_to_parking_lanes;
+        self.num_spots_per_offstreet = new.num_spots_per_offstreet;
+        self.driving_to_offstreet = new.driving_to_offstreet;
+        self.num_spots_per_lot = new.num_spots_per_lot;
+        self.driving_to_lots = new.driving_to_lots;
+
+        // For every spot filled or reserved before, make sure that same spot still exists. If not,
+        // evict that car.
+        let mut evicted = Vec::new();
+        for spot in filled_before {
+            if !avail_after.contains(&spot) {
+                let car = self.occupants.remove(&spot).unwrap();
+                evicted.push(self.parked_cars.remove(&car).unwrap());
+            }
+        }
+
+        // TODO How do we handle reserved_spots?
```

Comment:
Just some context: Spots are "reserved" when a car enters the beginning of the driving lane that's next to their intended parking spot. There were situations before where two cars would start parking in the same spot at almost the same time.